### PR TITLE
Flesh out core styles

### PR DESCRIPTION
**Pull request by Pinpickle**
Depends on #29

This fleshes out some of the empty placeholders left in the BEM PR, and adds some well-needed defaults (normalize.css, sensible typography)

Is still by no means pretty, we'll get there.

image

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ gulp.task('copy', () =>
 
 // Process the Stylus files into CSS
 gulp.task('styles', () =>
-  gulp.src('static/src/styles/**/*.styl')
+  gulp.src('static/src/styles/main.styl')
```
Comment (jaredkhan) [inline gulpfile.js]: Won't this only pick up changes to main and not recompile changes in other files?

File: static/src/styles/core/fonts.styl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// This file must be imported before any files that contain CSS output
+// This is because @import declarations must be the first declarations in CSS files
```
Comment (jaredkhan) [inline static/src/styles/core/fonts.styl]: With this in mind, is there a better way we should be importing fonts or is this fine?

File: static/src/styles/core/layout.styl [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+body {
+  background: $Color__lightBackground;
+}
+
+* {
+  box-sizing: border-box;
+}
+
+// We are violateing BEM naming rules here purely because this function is going to be used a lot
```
Comment (jaredkhan) [inline static/src/styles/core/layout.styl]: Bro do you even spell
`violateing`

Comment (Pinpickle) [inline static/src/styles/core/layout.styl]: It's 🥝 spelling I swear

File: gulpfile.js (hunk 1 above)
Comment (Pinpickle) [inline gulpfile.js]: Nah, this doesn't affect the watch code (which is further down), a recompile is still triggered if any file is changed.

File: static/src/styles/core/fonts.styl (hunk 2 above)
Comment (Pinpickle) [inline static/src/styles/core/fonts.styl]: Putting the `link` directly in our HTML will prevent us from accidentally breaking the fonts

File: gulpfile.js (hunk 1 above)
Comment (jaredkhan) [inline gulpfile.js]: Can see that this works but I don't understand why. My understanding is that `gulp.src` will output all files which match the given glob which in this case is just this one file.
I thought `changed(dest)` just filtered out its input to those files that have changed, how is it finding the changed files?

**Comment by jaredkhan** [comment]
Jeeeez. Dunno what happened to this merge.

**Comment by jaredkhan** [comment]
@Pinpickle Did you rebase a commit that was already merged 👹

File: gulpfile.js (hunk 1 above)
Comment (Pinpickle) [inline gulpfile.js]: I believe the `changed` isn't doing anything in this case as by default it compares hashes of files, not modification dates. `main.styl` will always be different to `main.css`. I'm also not sure if it knows to change file extensions like that by default, either.

**Comment by jaredkhan** [comment]
This has been merged into master, apologies for the confusion

**Pull request closed without merging**